d2bs/kolbot/libs/modules/AutoEquip.js

```diff
@@ -313,38 +31,59 @@
 		return !!dep;
 	};
 
-	function AutoEquip() { // So we can call new upon it. Not sure why yet
+	const _defaults = {
+		reference: me,
+	};
 
+	/**
+	 * @constructor
+	 */
+	function AutoEquip(settings) { // So we can call new upon it. Not sure why yet
+		Object.assign(this, settings, _defaults);
 	}
 
-	AutoEquip.want = function (item) {
-		return item.__wanted__by_AutoEquip = (function () {
+	/**
+	 * @param {Item[]} items
+	 * @returns Item[]
+	 */
+	AutoEquip.prototype.compareRetAll = function (items) {
+		return items.sort(sortItems.bind(this));
+	};
+
+	AutoEquip.prototype.compare = function (...args) {
+		return this.compareRetAll(args).first();
+	};
+
+	AutoEquip.prototype.want = function (item) {
+		let key = '__wanted__by_AutoEquip' + this.id;
+		return item[key] = (function () {
 			if (!item) return false; // We dont want an item that doesnt exists
+			if (!this.reference) return false; // If we dont exists (in case of a merc)
 
 			// no quest items
-			if (['msf','vip'].includes(item.code)) return false;
+			if (['msf', 'vip'].includes(item.code)) return false;
 
 			// If we already excluded this item, lets not rerun this
-			if (item.hasOwnProperty('__wanted__by_AutoEquip') && !item.__wanted__by_AutoEquip) return false;
+			if (item.hasOwnProperty('key') && !item[key]) return false;
 
 			const bodyLoc = item.getBodyLoc();
```

Comment:
It does consider only player items location.
Item.getBodyLocation needs to be adjusted for each merc class as they differ.